pymc3/distributions/continuous.py

```diff
@@ -113,6 +129,9 @@ def __init__(self, *args, **kwargs):
         super(Flat, self).__init__(*args, **kwargs)
         self.median = 0
 
+    def random(self, point=None, size=None):
+        return np.zeros(size)
```

Comment:
Yeah, value error seems good to me.

On Sun, Jul 26, 2015 at 10:02 PM Abraham Flaxman <notifications@github.com>
wrote:

> In pymc3/distributions/continuous.py
> <https://github.com/pymc-devs/pymc3/pull/784#discussion_r35505801>:
>
> > @@ -113,6 +129,9 @@ def __init__(self, *args, **kwargs):
> >          super(Flat, self).__init__(*args, **kwargs)
> >          self.median = 0
> >
> > +    def random(self, point=None, size=None):
> > +        return np.zeros(size)
>
> Or, perhaps, a raise a ValueError, since it is an improper prior.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pymc-devs/pymc3/pull/784/files#r35505801>.
>